mlrun/serving/routers.py

```diff
@@ -157,3 +163,256 @@ def _handle_event(self, event):
+            return events
+        predictions = [
+            model.body["outputs"]["prediction"] for _, model in events.body.items()
+        ]
+
+        flattened_predictions = [
+            [predictions[j][i] for j in range(len(predictions))]
+            for i in range(len(predictions[0]))
+        ]
+        if self.vote_flag is None:
+            if all(
+                [
+                    all(res)
+                    for res in [
+                        list(map(self._is_int, prediction_arr))
+                        for prediction_arr in predictions
+                    ]
+                ]
+            ):
+                self.vote_type = "classificartion"
+            else:
+                self.vote_type = "regression"
+            self.vote_flag = True
+        if self.vote_type == "classification":
+            flattened_predictions = list(map(int, flattened_predictions))
+            result = self._max_vote(flattened_predictions)
+        else:
+            result = self._mean_vote(flattened_predictions)
+
+        event = {
+            "model_name": self.name,
+            "outputs": {
+                "id": events.body[list(events.body.keys())[0]].body["id"],
+                "inputs": events.body[list(events.body.keys())[0]].body["outputs"][
+                    "inputs"
+                ],
+                "prediction": result,
+            },
+            "id": events.body[list(events.body.keys())[0]].body["id"],
+        }
+
+        events.body = event
+        return events
+
+    def do_event(self, event, *args, **kwargs):
+        """handle incoming events, event is nuclio event class"""
+        start = datetime.now()
+        method = event.method or "POST"
+        if event.body and method != "GET":
+            event.body = self.parse_event(event)
+        urlpath = getattr(event, "path", "")
+
+        # if health check or "/" return Ok + metadata
+        if method == "GET" and (
+            urlpath == "/" or urlpath.startswith(self.health_prefix)
+        ):
+            setattr(event, "terminated", True)
+            event.body = self.get_metadata()
+            return event
+
+        # check for legal path prefix
+        if urlpath and not urlpath.startswith(self.url_prefix) and not urlpath == "/":
+            raise ValueError(
+                f"illegal path prefix {urlpath}, must start with {self.url_prefix}"
+            )
+
+        request = self._pre_event_processing_actions(event)
+        response = self.postprocess(self._vote(self._handle_event(event)))
+        if self._model_logger and self.log_router:
+            self._model_logger.push(start, request, response.body)
+        return response
+
+    def _parallel_run(self, event, how="thread"):
+        if how == "array":
+            results = {
+                model_name: model.run(copy.deepcopy(event))
+                for model_name, model in self.routes.items()
+            }
+        else:
```

Comment:
```suggestion
        elif mode == ParallelRunModes.thread.value:
```
and then in the end add:
```
else:
    raise NotImplementedError("Unknown parallel run mode")
```